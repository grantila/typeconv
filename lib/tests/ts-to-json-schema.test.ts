import { expect, describe, it } from "@jest/globals"

import {
	makeConverter,
	getTypeScriptReader,
	getJsonSchemaWriter,
} from '../index.js'


describe( 'ts-to-json-schema', ( ) =>
{
    // https://github.com/grantila/typeconv/issues/15
    // https://github.com/grantila/typeconv/issues/35
	it( 'interface heritage', async ( ) =>
	{
		const input = `
			interface Bar {
				b: string;
			}

			export interface Foo extends Bar {
				f: number;
			}
		`;

		const { convert } = makeConverter(
			getTypeScriptReader( ),
			getJsonSchemaWriter( ),
            {
                mergeObjects: true,
            }
		);

		const { data } = await convert( { data: input } );

		expect( data ).toMatchSnapshot( );
	} );

    // https://github.com/grantila/typeconv/issues/35
	it( 'type intersection', async ( ) =>
	{
		const input = `
			type Bar = {
				b: string;
			}

			export type Foo = Bar & {
				f: number;
			}
		`;

		const { convert } = makeConverter(
			getTypeScriptReader( ),
			getJsonSchemaWriter( ),
            {
                mergeObjects: true,
            }
		);

		const { data } = await convert( { data: input } );

		expect( data ).toMatchSnapshot( );
	} );
} );
