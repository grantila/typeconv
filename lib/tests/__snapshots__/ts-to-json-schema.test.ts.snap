// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ts-to-json-schema interface heritage 1`] = `
"{
  "definitions": {
    "Foo": {
      "type": "object",
      "properties": {
        "b": {
          "type": "string",
          "title": "Bar.b"
        },
        "f": {
          "type": "number",
          "title": "Foo.f"
        }
      },
      "required": [
        "b",
        "f"
      ],
      "additionalProperties": false,
      "title": "Foo, Bar"
    },
    "Bar": {
      "type": "object",
      "properties": {
        "b": {
          "type": "string",
          "title": "Bar.b"
        }
      },
      "required": [
        "b"
      ],
      "additionalProperties": false,
      "title": "Bar"
    }
  },
  "$comment": "Generated by core-types-json-schema (https://github.com/grantila/core-types-json-schema) on behalf of typeconv (https://github.com/grantila/typeconv)"
}"
`;

exports[`ts-to-json-schema type intersection 1`] = `
"{
  "definitions": {
    "Foo": {
      "type": "object",
      "properties": {
        "b": {
          "type": "string",
          "title": "Bar.b"
        },
        "f": {
          "type": "number",
          "title": "f"
        }
      },
      "required": [
        "b",
        "f"
      ],
      "additionalProperties": false,
      "title": "Foo, Bar"
    },
    "Bar": {
      "type": "object",
      "properties": {
        "b": {
          "type": "string",
          "title": "Bar.b"
        }
      },
      "required": [
        "b"
      ],
      "additionalProperties": false,
      "title": "Bar"
    }
  },
  "$comment": "Generated by core-types-json-schema (https://github.com/grantila/core-types-json-schema) on behalf of typeconv (https://github.com/grantila/typeconv)"
}"
`;
